[DOC] Composability of different threading runtimes (#26950)
openvinotoolkit/openvino: docs/articles_en/learn-openvino/llm_inference_guide/llm-inference-hf.rst

Now the model can be converted to OpenVINO using Optimum Intel Python API or CLI interfaces
mentioned above.

Execution on CPU device
##########################

As mentioned on :ref:`Inference threads wait actively <Inference_threads_wait_actively>`, OpenVINO default threading library

> oneTBB keeps CPU cores actively for 1ms after inference done. When using Optimum Intel Python API,

Reviewer: Same: 1ms is very specific to particular library - would avoid detailed numbers

> As mentioned on :ref:`Inference threads wait actively <Inference_threads_wait_actively>`, OpenVINO default threading library
> oneTBB keeps CPU cores actively for 1ms after inference done. When using Optimum Intel Python API,

Author: ```suggestion
As mentioned on :ref:`Composability of different threading runtimes <Composability_of_different_threading_runtimes>`, OpenVINO default threading runtime
oneTBB keeps CPU cores actively for a while after inference done. When using Optimum Intel Python API,
```

it will call Torch (via HF transformers) for postprocessing (for example beam search or gready search).
Torch uses OpenMP for threading, OpenMP will need to wait for CPU cores which are being kept actively by
oneTBB. OpenMP by default has the `busy-wait <https://gcc.gnu.org/onlinedocs/libgomp/GOMP_005fSPINCOUNT.html>`__ which can delay the next OpenVINO inference as well.

The recommendation

> * Limit the CPU core number used by Torch. `torch.set_num_threads <https://pytorch.org/docs/stable/generated/torch.set_num_threads.html>`__

Author: ```suggestion
* Limit the CPU thread number of Torch. `torch.set_num_threads <https://pytorch.org/docs/stable/generated/torch.set_num_threads.html>`__
```
